src/shop/ordering/orders.js

```diff
@@ -396,6 +545,278 @@ bloombox.shop.order.Order.prototype.addItem = function(item) {
+    'Invalid underlying order type value: "' + type + '".');
+};
+
+
+/**
+ * Inflate a wire order status into a JS SDK order status.
+ *
+ * @param {proto.opencannabis.commerce.OrderStatus|string|number} status Proto
+ *        status from an order to inflate into a JS SDK order status.
+ * @return {bloombox.shop.order.Status} Inflated order status.
+ * @throws {bloombox.shop.order.OrderException} If the status provided is not
+ *         valid or cannot be identified.
+ * @package
+ */
+bloombox.shop.order.Order.inflateStatus = function(status) {
+  if (status === null || status === undefined)
+    throw new bloombox.shop.order.OrderException(
+      'Invalid underlying order status value: "' + status + '".');
+  switch (status) {
+    case 'PENDING': return bloombox.shop.order.Status.PENDING;
+    case 'APPROVED': return bloombox.shop.order.Status.APPROVED;
+    case 'REJECTED': return bloombox.shop.order.Status.REJECTED;
+    case 'ASSIGNED': return bloombox.shop.order.Status.ASSIGNED;
+    case 'EN_ROUTE': return bloombox.shop.order.Status.EN_ROUTE;
+    case 'FULFILLED': return bloombox.shop.order.Status.FULFILLED;
+    case 0: return bloombox.shop.order.Status.PENDING;
+    case 1: return bloombox.shop.order.Status.APPROVED;
+    case 2: return bloombox.shop.order.Status.REJECTED;
+    case 3: return bloombox.shop.order.Status.ASSIGNED;
+    case 4: return bloombox.shop.order.Status.EN_ROUTE;
+    case 5: return bloombox.shop.order.Status.FULFILLED;
+  }
+  bloombox.logging.warn('Unable to resolve order status with value "' +
+    status + '". Defaulting to "PENDING".');
+  return bloombox.shop.order.Status.PENDING;
+};
+
+
+/**
+ * Inflate a shop order from its underlying proto object.
+ *
+ * @param {proto.opencannabis.commerce.Order} protob Commercial order object.
+ * @return {bloombox.shop.order.Order} Inflated SDK order object.
+ * @package
+ */
+bloombox.shop.order.Order.fromProto = function(protob) {
+  // @TODO: decode additional properties here once they are available
+  let targetId = protob.getId();
+  let underlyingOrderType = protob.getType();
+  let underlyingStatus = protob.getStatus();
+
+  let status = bloombox.shop.order.Order.inflateStatus(underlyingStatus);
+  let type = bloombox.shop.order.Order.inflateType(underlyingOrderType);
+  let order = new bloombox.shop.order.Order(type, null, null);
+  order.setId(targetId);
+  order.setStatus(status);
+
+  return order;
+};
+
+
+/**
+ * Retrieve a copy of this order from the server, and update the properties on
+ * it accordingly.
+ *
+ * @param {bloombox.shop.OrderGetCallback} callback Callback to dispatch once
+ *        the order update has completed.
+ * @throws {bloombox.shop.order.OrderException} If no order ID is specified on
+ *        the current order, indicating it has not successfully been submitted
+ *        to the shop service yet.
+ * @export
+ */
+bloombox.shop.order.Order.prototype.update = function(callback) {
```

Comment:
method to update an order's data from the server:
`order.update(function(error, updatedOrder) { ... });`